stdlib/src/builtin/simd.mojo

```diff
@@ -513,6 +508,19 @@ struct SIMD[type: DType, size: Int = simdwidthof[type]()](
             rebind[Scalar[type]](self).value
         )
 
+    @always_inline("nodebug")
+    fn __index__(self) -> Int:
+        """Returns the value as an int if it is an integral value
+
+        Contraints:
+            Must be an integral value
+
+        Returns:
+            The value as an integer
+        """
+        constrained[type.is_integral(), "expected integral type"]()
```

Comment:
NOICE

Reply:
This may not be a good idea, for `SIMD` at least. This way, `SIMD[bool, 1]` will be cast to a `Int` but we might want to index with the original mask (which is size 1).

Reply:
> This may not be a good idea, for `SIMD` at least. This way, `SIMD[bool, 1]` will be cast to a `Int` but we might want to index with the original mask (which is size 1).

Now that you mention it `Scalar[bool]` actually won't be accepted here, which is inconsistent as I've currently given `Bool` the `Indexer` trait, but I was hoping for feedback from the team on whether that was acceptable or not.

We need some support for this in SIMD, however, since it would be expected that one could use any of the integral scalars to index the stdlib containers.

Reply:
I just think we need a more complete design before adding the support to stdlib.